### ISSUE TITLE
Import status dashboard pagination issue - follow-up to #4317

### ISSUE THREAD

**Issue by stefan-korn** [GetDKAN/dkan#4332]
### Current Behavior

since #4317 the pagination of the import status dashboard is not correct anymore, if there are datasets filtered out due to non-importable distributions. Pagination is still pointing to all datasets irrespective of the filter. This may lead to empty pages in the pager and pages with less items than pagination items per page setting.

### Expected Behavior

Pager should reflect the actual state of filtered datasets.

### Steps To Reproduce

Have more than 10 datasets, some of them having no importable distributions (like PDF, DOC, etc.).
See that pagination of import status dashboard is not correct.

**Comment by dafeder** [comment]
@stefan-korn I agree this needs to be fixed but yeah, this is closer to the way the dashboard worked once upon a time and the performance was terrible. I think we need a new approach, something where we query for _resources_ and can filter them by whether they're store-able or not, and then get any metadata we need for them. I'm going to keep this issue open and hope to come back to it soon. Unless you have some evidence that the performance issues can be mitigated (maybe through caching of the DatasetInfo objects if we can get the invalidation right) I think we may close the PR.

**Comment by kaise-lafrai** [comment]
@janette @dafeder @paul-m 

I was able to replicate the issue reported. The dashboard performs a query to grab a count of all data node types. The total is then used to build the pagination. I tried looking into how the count is calculated and tracked it down to dkan/modules/metastore/src/Storage/Data.php , see code below. My initial thought was that we could create a new count that would avoid counting the distributions with a mimetype of something like a .pdf. Then it can return an appropriate count and the pager would work as intended in theory. But this turned out to be a mess, with more time I could probably get it to work. I’m interested in hearing what you and Dan think of that approach.
```
/**
   * {@inheritdoc}
   */
  public function count($unpublished = FALSE): int {
    return $this->listQueryBase(NULL, NULL, $unpublished)->count()->execute();
  }
```
Here are a few options to address this issue in the meantime:

- Revert some of the dashboard changes that were introduced [here](https://github.com/GetDKAN/dkan/pull/4317). Don’t hide the resources and include a new fetch/store/post import status that reflects the appropriate handling of the resource. Maybe a status of “Unimportable Resource”

- Look into adding a filter to hide/display resources that are unimportable. (Not sure if this is possible and would need to look into it some more, it might not resolve the pagination issue.)

**Comment by dafeder** [comment]
I think we talked about an option for changing the styling of the row, possibly adding a span so that we don't have to have a fake status. Just a greyed out row maybe that goes across and rather than individual status columns just says something like "Resource not compatible with datastore". @stefan-korn how would that work for you at least as a stopgap?

**Comment by stefan-korn** [comment]
I was just noticing the wrong pagination here and issues that went along with that (like empty pages). So as long as the pagination works or at least these issues are gone, I would probably be fine with any kind of display of the dashboard.

**Comment by kaise-lafrai** [comment]
@dafeder That idea seems fine to me. Either way we would continue to display all data in the dashboard if we decide to change it. I would like to get @janette opinion on it and maybe a UX/UI person to confirm best practices around accessibility.
